pandas/plotting/_matplotlib/core.py

```diff
@@ -934,6 +934,8 @@ def __init__(self, data, x, y, s=None, c=None, **kwargs):
             # hide the matplotlib default for size, in case we want to change
             # the handling of this argument later
             s = 20
+        elif s in data.columns:
```

Comment:
This might fails for things like `s=[1, 2, 3]`, non-hashable inputs. We should have other cases where we handle arrays or strings. In `_make_plot` we have something like

```
c_is_column = is_hashable(c) and c in self.data.columns
```

Something similar may be required here.

Reply:
These are some notes for myself.

According to matplotlib (https://matplotlib.org/3.1.1/api/_as_gen/matplotlib.pyplot.scatter.html), s is `scalar or array_like, shape (n, )`. So the decision tree for `s` is:
- None? If yes, then set s=20 (as it was defined in the code before), if no continue checking s.
- Scalar (`np.ndim(s) == 0`)? If yes, then proceed with the passed value for `s`, else continue checking.
- Hashable and in columns? (`is_hashable(s) and s in data.columns`) If yes, then set `s = data[s]`, else continue checking `s`.
- If this is a list, then check that the list is as long as the number of points to plot (`np.ndim(s)==1 and len(s)==len(x)`) and pass it, else proceed (which will trigger matplotlib errors).

Reply:
> This might fails for things like `s=[1, 2, 3]`, non-hashable inputs. We should have other cases where we handle arrays or strings. In `_make_plot` we have something like
> 
> ```
> c_is_column = is_hashable(c) and c in self.data.columns
> ```
> 
> Something similar may be required here.

Could you elaborate on the strings? (I'm not sure what is an example of this.)